package.json

```diff
@@ -5,10 +5,24 @@
   "scripts": {
     "prettier": "prettier --write .",
     "update": "node update-dep.mjs",
-    "new-example": "plop example"
+    "new-example": "plop example",
+    "lint-staged": "lint-staged",
+    "prepare": "husky install"
   },
+  "pre-commit": "lint-staged",
   "devDependencies": {
-    "prettier": "^2.5.1",
-    "plop": "^3.0.5"
+    "eslint": "^8.9.0",
+    "husky": "^7.0.4",
+    "lint-staged": "^12.3.4",
+    "plop": "^3.0.5",
+    "prettier": "^2.5.1"
+  },
+  "lint-staged": {
+    "**/*.{js,jsx,ts,tsx}": [
+      "prettier --write"
```

Comment:
There's a reason for this separation, this stage will include eslint too, but I'll add it in another PR because some examples don't have eslint setup.